test/scheduler_test.rb

```diff
@@ -456,6 +456,52 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      def self.before_enqueue_example(*)
+        if Resque.redis.setnx('before_enqueue_job:enqueued', 'true')
+          sleep 5 # sleep longer than test timeout
+          true
+        else
+          false # enqueue only once
+        end
+      end
+
+      def self.perform(_, _)
+      end
+    end
+
+    schedule = {
+      'BeforeEnqueueJob' => { cron: '* * * * * *', class: 'BeforeEnqueueJob' }
+    }
+
+    Resque::Scheduler.enqueue_from_config(schedule['BeforeEnqueueJob'])
+
+    pid = fork do
+      Resque.schedule = schedule
+      Resque::Scheduler.run
+    end
+
+    begin
+      Timeout.timeout(3) do
+        loop do
+          break if Resque.redis.get('before_enqueue_job:enqueued') == 'true'
+          sleep 0.1
+        end
+      end
+    rescue Timeout::Error
+      flunk 'Child process did not terminate in time'
+    ensure
+      Process.kill('TERM', pid)
+      Process.wait(pid)
+    end
+
+    assert_equal 'true', Resque.redis.get('before_enqueue_job:enqueued')
+    assert_equal 1, Resque.size('quick')
+  end
```

Comment:
I tried reverting the lib changes and running this test and it passed.  It looks like you aren't properly checking that the job actually gracefully shutdown.  Maybe you need to do something in perform and assert that it was done at the end of the test